src/nncf/onnx/graph/passes.py

```diff
@@ -77,3 +79,56 @@ def apply_preprocess_passes(model: onnx.ModelProto) -> onnx.ModelProto:
     # Otherwise, not all no-op Cast nodes will be found.
     preprocessed_model = eliminate_nop_cast(preprocessed_model)
     return preprocessed_model
+
+
+def compress_quantize_weights_transformation(model: onnx.ModelProto):
+    """
+    Transforms the model by folding `QuantizeLinear` nodes with constant inputs
+    into precomputed, quantized initializers.
+
+    This transformation finds `QuantizeLinear` nodes with constant inputs
+    (i.e., inputs present in the model's initializers), precomputes their quantized values,
+    updates the initializer with these results, and removes the corresponding
+    `QuantizeLinear` nodes from the graph.
+
+    :param model: The model to be transformed.
+    """
+    initializer = {x.name: x for x in model.graph.initializer}
+    nodes_to_remove = []
+
+    version = max(model.opset_import[0].version, 19)
+    QuantizeLinear = load_op("", "QuantizeLinear", version)
+
+    for node in model.graph.node:
+        if node.op_type != "QuantizeLinear":
+            continue
+
+        x_name, y_scale_name, y_zero_point_name = node.input[:3]
```

Comment:
Will zero point always be present?

Reply:
Good catch! According to the [documentation](https://onnx.ai/onnx/operators/onnx__QuantizeLinear.html#inputs), the QuantizeLinear may contain between 2 and 3 inputs. I'll fix it.